example/build_cmake_installed/CMakeLists.txt

```diff
@@ -16,6 +16,7 @@ endif()
 find_package(Kokkos REQUIRED)
 
 add_executable(example cmake_example.cpp foo.f)
+set_target_properties(example PROPERTIES LINKER_LANGUAGE Fortran)
```

Comment:
For  ECP members: This is https://gitlab.kitware.com/ecp/cmake-support/-/issues/20

Reply:
/CC @bradking

Reply:
Ok, this should be fixed in CMake-3.24.

Reply:
CMake 3.24 adds support for LLVM/Flang 15+. LLVM/Flang 14 and below do not have everything needed.  See [CMake Issue 22387](https://gitlab.kitware.com/cmake/cmake/-/issues/22387).

Reply:
@bradking Even with cmake-3.24.1 and clang/flang 15.0.0, we still seem to need that line above:
```
Run cmake -B builddir -DCMAKE_CXX_COMPILER=clang++
-- The CXX compiler identification is Clang
-- The Fortran compiler identification is LLVMFlang 15.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib64/ccache/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /usr/bin/flang-new - skipped
-- Found OpenMP_CXX: -fopenmp=libomp (found version "5.0") 
-- Found OpenMP_Fortran: -fopenmp (found version "3.0") 
-- Found OpenMP: TRUE (found version "5.0")  
-- Enabled Kokkos devices: OPENMP
-- Configuring done
-- Generating done
-- Build files have been written to: /__w/kokkos/kokkos/example/build_cmake_installed/builddir
Scanning dependencies of target example
[ 33%] Building CXX object CMakeFiles/example.dir/cmake_example.cpp.o
[ 66%] Building Fortran object CMakeFiles/example.dir/foo.f.o
[100%] Linking CXX executable example
/usr/bin/ld: CMakeFiles/example.dir/foo.f.o: in function `print_fortran_':
/__w/kokkos/kokkos/example/build_cmake_installed/foo.f:2: undefined reference to `_FortranAioBeginExternalListOutput'
/usr/bin/ld: /__w/kokkos/kokkos/example/build_cmake_installed/foo.f:2: undefined reference to `_FortranAioOutputAscii'
/usr/bin/ld: /__w/kokkos/kokkos/example/build_cmake_installed/foo.f:2: undefined reference to `_FortranAioEndIoStatement'
clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [CMakeFiles/example.dir/build.make:11[9](https://github.com/kokkos/kokkos/actions/runs/3191601038/jobs/5208106816#step:16:10): example] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/example.dir/all] Error 2
gmake: *** [Makefile:[10](https://github.com/kokkos/kokkos/actions/runs/3191601038/jobs/5208106816#step:16:11)1: all] Error 2
Error: Process completed with exit code 2.
```
(see https://github.com/kokkos/kokkos/actions/runs/3191601038/jobs/5208106816)

Reply:
Make sure you have `FFLAGS=-flang-experimental-exec` in the environment.  Otherwise `flang` cannot link binaries and CMake's detection of implicit link libraries might not work.

Reply:
@dalg24 do you want to add this flag?

Reply:
> @dalg24 do you want to add this flag?

Not sure what you are asking.
Do I want to implement the changes?  No
Do I have any issue with Brad's suggestion?  No

Reply:
I added the flag, which will allow us to drop the CMake change in the other files.

Reply:
Please attach `CMakeFiles/*/CMakeFortranCompiler.cmake`.

Reply:
@bradking https://gitlab.kitware.com/cmake/cmake/-/merge_requests/7781 fixed the undefined reference, when using `clang++` & `flang-new`. Also `-flang-experimental-exec` isn't needed.